tests/garage/tf/models/test_new_model_pickle.py

```diff
@@ -0,0 +1,63 @@
+import pickle
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+
+
+class TestNewModelPickling(unittest.TestCase):
+    def test_model_pickle(self):
+        sess = tf.Session()
+        with sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model = GaussianMLPModel(output_dim=2)
+            model.build(input_var)
+            data = np.random.random((3, 5))
+            results = sess.run(model.outputs[:2], feed_dict={input_var: data})
+            model_data = pickle.dumps(model)
+
+        sess.close()
+        tf.reset_default_graph()
+
+        new_sess = tf.Session()
+        with new_sess as sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model_pickled = pickle.loads(model_data)
+            model_pickled.build(input_var)
+
+            results2 = sess.run(
+                model_pickled.outputs[:2], feed_dict={input_var: data})
```

Comment:
I talked about this in the old model [pr](https://github.com/rlworkgroup/garage/pull/481#discussion_r252490100), I think making this a namedtuple is better. It is less required if a user often calls outputs instead of part of it.

Reply:
i agree. can we make that a feature of `Model`?

Reply:
or a feature of `Network`? So that we have `model.network_1.mean_var`, etc.

Reply:
a feature of network